aten/src/ATen/native/cpu/SortingKernel.cpp

```diff
@@ -69,7 +73,8 @@ void _dim_apply(
         }
       };
 
-      iter.for_each(loop);
+      int64_t grain_size = internal::GRAIN_SIZE / std::max(int64_t{1}, dim_size);
+      iter.for_each(loop, /*grain_size=*/grain_size);
```

Comment:
If `GRAIN_SIZE < dim_size`, wouldn't it result in zero? Also, wouldn't having smaller grain size for larger dimension be less performant? And if I am missing something, could you please insert a comment on why you choose this specific grain size?

Reply:
grain_size=0 is OK, my goal is to increase the chance of parallelization by decreasing the grain_size, and the grain_size setting refers to topk.

Reply:
The sorting along a specific dimension is done by a single thread. I would assume that the grain size has to actually depend on `prod(shape[:dim] + shape[dim+1:])`, because this is exactly the amount of elements distributed among the threads?